src/hotspot/os/linux/systemMemoryBarrier_linux.cpp

```diff
@@ -61,6 +61,18 @@ static long membarrier(int cmd, unsigned int flags, int cpu_id) {
 }
 
 bool LinuxSystemMemoryBarrier::initialize() {
+#if defined(RISCV)
+// RISCV port was introduced in kernel 4.4.
+// 4.4 also made membar private expedited mandatory.
+// But RISCV actually don't support it until 6.9.
+  long major, minor;
+  os::Linux::kernel_version(&major, &minor);
+  if (!(major >= 6 && minor >= 9)) {
```

Comment:
Is there something wrong with this conditional expression? What happens if it's 7.0?

Reply:
Yes, thanks!